lib/Routing/Dynamic/DocumentRouteHandler.php

```diff
@@ -103,11 +105,21 @@ public function getRouteByName(string $name): ?DocumentRoute
     public function matchRequest(RouteCollection $collection, DynamicRequestContext $context): void
     {
         $document = Document::getByPath($context->getPath());
+        $site = $this->siteResolver->getSite($context->getRequest());
+
+        // If the request is not from a site and the document is part of a site
+        // or the ID of the requested site does not match the site where the document is located.
+        // Then we have to throw a NotFoundHttpException
+        if(!$site && $document && !$context->getRequest()->get('pimcore_editmode')) {
```

Comment:
Maybe call `Tool::isFrontendRequestByAdmin()` instead of just checking for editmode. otherwise it broken for other features in admin 
<img width="439" alt="image" src="https://user-images.githubusercontent.com/5137917/228196151-d2b58b23-6de5-405c-a7bd-d11d49a0ec66.png">